### PR TITLE
test: migrate BulkActionsToolbar tests

### PR DESCRIPTION
**Pull request by PavelKirvelTextkernel**
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. Use imperative, present tense in your commit description ("change", not "changed" or "changes") without uppercases or period (.) at the end.

Task Link: https://textkernel.atlassian.net/browse/ONEUI-354

### CODE REVIEW

File: src/components/BulkActionsToolbar/__tests__/BulkActionsToolbar.spec.tsx [hunk 1]
```diff
@@ -88,14 +92,15 @@ describe('BulkActionsToolbar component', () => {
                 actions={actions}
             />
         );
-        wrapper.find('.BulkActionsToolbar__toggleButton').simulate('click');
+        await userEvent.click(screen.getByRole('button', { name: 'Select All' }));
```
Comment (eszthoff) [inline src/components/BulkActionsToolbar/__tests__/BulkActionsToolbar.spec.tsx:95]: I also saw it in the other [PR](https://github.com/textkernel/oneui/pull/949#discussion_r1178133030). It is not clear to me when we need/should initialise a user with `userEvent.setup()`. Can you look into it, and provide some clarity around it? Thanks

File: src/components/BulkActionsToolbar/__tests__/BulkActionsToolbar.spec.tsx [hunk 2]
```diff
@@ -124,19 +130,18 @@ describe('BulkActionsToolbar component', () => {
...
+        await userEvent.click(screen.getByRole('button', { name: 'Import' }));
+        await userEvent.click(screen.getByRole('button', { name: 'Delete' }));
 
-        // clicking on disabled "Compare" action
-        wrapper.find('.BulkActionsToolbar__action').at(0).simulate('click');
-        // clicking on enabled "Import" action
-        wrapper.find('.BulkActionsToolbar__action').at(1).simulate('click');
-        // clicking on enabled "Delete" action
-        wrapper.find('.BulkActionsToolbar__action').at(2).simulate('click');
-        expect(onCompareClick).toHaveBeenCalledTimes(1);
```
[the remark this answers is not in the document]
Comment (eszthoff) [reply src/components/BulkActionsToolbar/__tests__/BulkActionsToolbar.spec.tsx:134]: nice catch. Can you add a comment in the code that this button is disabled?

File: src/components/BulkActionsToolbar/__tests__/BulkActionsToolbar.spec.tsx [hunk 3]
```diff
@@ -21,19 +23,19 @@ describe('BulkActionsToolbar component', () => {
...
         {
             label: 'Delete',
             disabled: false,
             context: 'primary',
             onClick: onDeleteClick,
-        },
+        } as BulkActionsToolbarAction,
     ];
 
     beforeEach(() => {
```
Comment (eszthoff) [inline src/components/BulkActionsToolbar/__tests__/BulkActionsToolbar.spec.tsx:41]: (note: this comment is for the next line, but I cannot add it there.)

same as https://github.com/textkernel/oneui/pull/949#discussion_r1178129921. I think we should improve these tests and remove such mocks.

File: src/constants/style-related.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-export const VALIDATION_CONTEXTS = ['success', 'warning', 'danger'] as const;
+export const VALIDATION_CONTEXTS = ['success', 'warning', 'danger', 'primary'] as const;
```
Comment (eszthoff) [inline src/constants/style-related.ts:1]: why is this change?

**Review by eszthoff — CHANGES_REQUESTED** [review]

File: src/components/BulkActionsToolbar/__tests__/BulkActionsToolbar.spec.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (PavelKirvelTextkernel) [reply src/components/BulkActionsToolbar/__tests__/BulkActionsToolbar.spec.tsx:134]: :+1:

File: src/constants/style-related.ts (hunk 4 above)
Comment (PavelKirvelTextkernel) [reply src/constants/style-related.ts:1]: We use Context type for context and this test case contains  primary for actions

File: src/components/BulkActionsToolbar/__tests__/BulkActionsToolbar.spec.tsx (hunk 1 above)
Comment (PavelKirvelTextkernel) [reply src/components/BulkActionsToolbar/__tests__/BulkActionsToolbar.spec.tsx]: I've checked some articles and only find that it recommened to use https://testing-library.com/docs/user-event/setup/

File: src/constants/style-related.ts (hunk 4 above)
Comment (PavelKirvelTextkernel) [reply src/constants/style-related.ts:1]: Should we add it or just change actions data ?

Comment (eszthoff) [reply src/constants/style-related.ts:1]: I'm not sure what test you referring to. This is a global constant that will effect all components using context, isn't it?

Comment (PavelKirvelTextkernel) [reply src/constants/style-related.ts:1]: @eszthoff yes these test actions 
image

Comment (PavelKirvelTextkernel) [reply src/constants/style-related.ts:1]: I can change it to warning or should we use here primary ?

Comment (eszthoff) [reply src/constants/style-related.ts:1]: I think this context has been removed with the color changes, and the test wasn't fixed. I like your suggestion to change the test to `warning`

Comment (PavelKirvelTextkernel) [reply src/constants/style-related.ts:1]: :+1:

File: src/components/BulkActionsToolbar/__tests__/BulkActionsToolbar.spec.tsx (hunk 3 above)
Comment (PavelKirvelTextkernel) [reply src/components/BulkActionsToolbar/__tests__/BulkActionsToolbar.spec.tsx]: :+1:

File: src/components/BulkActionsToolbar/__tests__/BulkActionsToolbar.spec.tsx [hunk 5]
```diff
@@ -58,12 +50,13 @@ describe('BulkActionsToolbar component', () => {
                 actions={actions}
             />
         );
-        expect(view.container).toMatchSnapshot();
 
+        expect(view.container).toMatchSnapshot();
         expect(consoleError).not.toHaveBeenCalled();
```
[the remark this answers is not in the document]
Comment (PavelKirvelTextkernel) [reply src/components/BulkActionsToolbar/__tests__/BulkActionsToolbar.spec.tsx]: :+1:

**Review by eszthoff — APPROVED** [review]

**Pull request merged**